### PR TITLE
Add the setting "confirmCloseAllTabs" to SUI

### LINKED ISSUE #14033: Option to disable dialog "Do you want to close all tabs?" like a checkbox in the dialog
Hello, I know it's possible to edit the json to add 
`"confirmCloseAllTabs": false,`

But it should be possible without having to search on the github issues to find there is a json setting available to do that.

It should be intuitive and easy to configure. Something like a checkbox in the dialog "never show this dialog again".

It's an issue from the UX point of view in my opinion.
Not hard to implement.

Anyway, thanks a lot. I lot Windows Terminal soooo much.

### LINKED ISSUE #14413: Add the setting "confirmCloseAllTabs" to the UI
# Description of the new feature/enhancement

 The setting "confirmCloseAllTabs" should be added to the ui so that it can be easily found.

# Proposed technical implementation details (optional)

### PR DESCRIPTION
**Pull request by leejy12**
This commit adds the setting "confirmCloseAllTabs" to SUI.
The setting was added to the Interactions pane of Global Settings.

Closes #14413
Closes #14033

### CODE REVIEW

**Comment by leejy12** [comment]
I think a better Header for the settings container is needed. Maybe a HelpText also.

**Comment by carlos-zamora** [comment]
FYI to reviewer, here's what it looks like:
Toggle switch displaying "Confirm close all tabs"

**Review by carlos-zamora — APPROVED** [review]
Verified this works with screen readers.

Long-term, we need...
1. "don't show this again" checkboxes on the warning dialogs. When checked, we suppress the dialog and save the selection to state.json.
2. Migration (aka legacy support): if user has any warning settings set in settings.json, update state.json appropriately.
3. SUI updates:
   - add a UI for each warning so that the user can reset it.
   - add a "reset all warnings" button? 

I'm saying all of this because adding these warnings to the SUI in the meantime shouldn't conflict with any of it. At the end of the day, if 1 and 2 is done, we just update the view model in SUI to update state.json instead of settings.json.

tl;dr: Thanks for doing this :). If you want to add all of the other warnings, go for it. Should be pretty much the same.

**Comment by zadjii-msft** [comment]
FWIW, Carlos's comment is largely musing and not indicative of work that NEEDS to be done in this PR. Small, incremental changes are always easier to review and ingest.

See also #6641

**Review by lhecker — APPROVED** [review]

**Comment by carlos-zamora** [comment]
> FWIW, Carlos's comment is largely musing and not indicative of work that NEEDS to be done in this PR. Small, incremental changes are always easier to review and ingest.
> 
> See also #6641

Sorry! Yes! Should've made that more clear haha. Just trying to be encouraging and give you (or anybody else) the direction we're thinking of going as a whole.

**Comment by DHowett** [comment]
For phrasing purposes, this might be better stated as, "Confirm when closing more than one tab". We cannot always translate the JSON names directly to sentences :smile:
